pkg/controllers/user/controllers.go

```diff
@@ -104,10 +106,18 @@ func RegisterFollower(ctx context.Context, cluster *config.UserContext, kubeConf
 }
 
 func RegisterUserOnly(ctx context.Context, cluster *config.UserOnlyContext) error {
-	if err := createUserClusterCRDs(ctx, cluster); err != nil {
+	err := createUserClusterCRDs(ctx, cluster)
+	if err != nil {
 		return err
 	}
 
+	// Don't register secret powered setting in the local cluster
+	if cluster.ClusterName != setting.LocalClusterName {
+		if err = settingprovider.Register(ctx, cluster); err != nil {
+			return err
+		}
+	}
+	ingresshostgen.Register(ctx, cluster)
```

Comment:
q: do we now need to change to read from our own setting provider? As in, update https://github.com/rancher/rancher/blob/60f8056a42ff532f4f025abeefe235144a1ebc84/pkg/controllers/user/ingresshostgen/ingresshostgen.go#L38 to read from https://github.com/rancher/rancher/blob/1ca9d02b9af7067e1612be982c47c6d838914889/pkg/controllers/user/settingprovider/register.go#L29:6

Reply:
nope, it will still work with existing `settings` as we have set the settings provider at https://github.com/rancher/rancher/pull/24138/files#diff-7c122d27326c29041670171d463b00ceR22 , so it will be powered by secret type settings automatically.